src/cli.zig

```diff
@@ -323,6 +363,35 @@ test "parse_size" {
     try expectEqual(@as(u64, 1000 * kib), parse_size("  1000  kb "));
 }
 
+/// Given a limit like `10GiB`, return the maximum power-of-two count of `T`s
+/// that can fit in the limit.
+fn parse_size_to_count(comptime T: type, string_opt: ?[]const u8, comptime default: u32) u32 {
+    comptime assert(default >= 2048);
+    comptime assert(math.isPowerOfTwo(default));
+
+    var result: u32 = undefined;
+    if (string_opt) |string| {
```

Comment:
Is this the the idiomatic way to spell "multiline ternary"?

Reply:
See https://github.com/tigerbeetledb/tigerbeetle/pull/282#discussion_r1031626351 — I try to use optionals instead of `undefined` when it isn't too complicated.

The `_opt` is fine, I think we haven't really settled on an idiom there.